ambry-api/src/main/java/com.github.ambry/config/StoreConfig.java

```diff
@@ -171,6 +171,17 @@
   @Default("240000")
   public final int storeStatsIndexEntriesPerSecond;
 
+  /**
+   * Specifies the minimum size that index entries should occupy when they get persisted. If the number of bytes for
+   * constituting keys and values fall short of this size, the entries will be padded with dummy bytes to amount to this
+   * number.
+   * Setting this value to N bytes ensures that even if the size of keys put to the store changes at runtime, as long as
+   * the total entry size is still N bytes, the key size change will not cause the active index segment to roll over.
+   */
+  @Config("store.index.persisted.entry.min.bytes")
```

Comment:
this only works with key size changes. Index value change may be accommodated in the future but it is not a requirement at this time, so kept things simple.
This only needs to be tweaked if we introduce a key that gets serialized along with its value to be more than 115. Currently, the maximum an entry can be is 115 bytes, so chose that as the min bytes here, so that once BlobId V2 becomes standard we won't waste any bytes in padding.